faunadb-java-dsl/src/main/java/com/faunadb/client/query/Language.java

```diff
@@ -163,189 +182,300 @@ public Path at(int... others) {
   }
 
   /**
-   * Creates a RefV value. The string "classes/widget/123" will be equivalent to:
+   * Creates a new reference.
+   * For example {@code Ref("classes/users/123")}.
    *
-   * <pre>
-   * {@code new RefV("123", new RefV("widget", Native.CLASSES)) }
-   * </pre>
+   * <p>
+   * The usage of this method is discouraged.
+   * Prefer {@link #Ref(Expr, String)} or {@link #Ref(Expr, Expr)}.
+   * </p>
    *
+   * @param ref the reference id
+   * @return a new {@link Expr} instance
    * @see <a href="https://fauna.com/documentation/queries#values">FaunaDB Values</a>
+   * @see #Ref(Expr, String)
    */
   public static Expr Ref(String ref) {
     return Fn.apply("@ref", Value(ref));
   }
 
   /**
-   * Calls ref function to create a ref value.
+   * Creates a new scoped reference.
+   * For example {@code Ref(Class("users"), "123")}.
    *
+   * @param classRef the scope reference. Type: Reference
+   * @param id the reference id. Type: String
+   * @return a new {@link Expr} instance
    * @see <a href="https://fauna.com/documentation/queries#values">FaunaDB Values</a>
+   * @see #Class(String)
+   * @see #Database(String)
+   * @see #Index(String)
+   * @see #Function(String)
    */
   public static Expr Ref(Expr classRef, Expr id) {
     return Fn.apply("ref", classRef, "id", id);
   }
 
   /**
-   * Calls ref function to create a ref value.
+   * Creates a new scoped reference.
+   * For example {@code Ref(Class("users"), "123")}.
    *
+   * @param classRef the scope reference. Type: Reference
+   * @param id the reference id
+   * @return a new {@link Expr} instance
    * @see <a href="https://fauna.com/documentation/queries#values">FaunaDB Values</a>
+   * @see #Class(String)
+   * @see #Database(String)
+   * @see #Index(String)
+   * @see #Function(String)
    */
   public static Expr Ref(Expr classRef, String id) {
     return Ref(classRef, Value(id));
   }
 
   /**
-   * Returns a native reference to all classes.
+   * Returns a reference to a set of all classes in the database.
+   * A reference set must be paginated in order to retrieve its values.
+   *
+   * @return a new {@link Expr} instance
+   * @see #Paginate(Expr) 
    */
   public static Expr Classes() {
     return Classes(Null());
   }
 
   /**
-   * Returns a native reference to all classes in a nested database.
+   * Returns a reference to a set of all in a nested database.
```

Comment:
I think it might be a little clear to not refer to nested databases, something like the following:

`Returns a reference to a set of all keys in the specified database (i.e. scope)`

there are several items which require this update:
Classes, Indexes, Keys, Functions,...